sympy/matrices/common.py

```diff
@@ -2224,6 +2225,19 @@ def __neg__(self):
 
     @call_highest_priority('__rpow__')
     def __pow__(self, exp):
+        return self.pow(exp)
+
+    def pow(self, exp, mulsimp=None, jordan=None):
+        """Return self**exp a scalar or symbol.
+
+        Parameters
+        ==========
+
+        mulsimp : bool, optional
+            Specifies whether intermediate term algebraic simplification is used
+            during recursive power evaluations to control expression blowup.
```

Comment:
The jordan parameter is undocumented here. Also I think that having `jordan=None` is not ideal in case another method is added in future. It would be better to have something like `method=mul/jordan/None`.

Reply:
It doesn't look like there are any tests for calling pow and asking for the Jordan form method

Reply:
I see 1.5 is out and nobody really picked this up and matrices are still blowing up, I have a block of free time so curious if you would want me to pick this up, make whatever small changes needed so it could be incorporated?

Reply:
Yes, sorry I forgot about this. I had been intending to submit a PR based on this myself to get it in but didn't find the time. If you are able to work on this then that would be great.

I would like this to be reviewed by someone who is more familiar than me with the design of the matrices code in SymPy but I think this PR is now quite daunting with the sheer number of messages and commits.

My suggestion would be (this is what I intended to do):
1. Squash the commits in some way
2. Rebase on to current master
3. Open a new PR and explain the rationale for the changes as they stand now.

Then I think it would be easier for someone to review. The main questions I have are about which classes this should be added to and whether the API makes sense with the direction of travel for the matrices module and I would like someone else to answer those questions.

One other point I noticed when looking at this some time ago was that the tests were quite slow. I recall thinking that it wasn't necessary to check so many permutations of matrix types with such large matrices (or something like that...).

Reply:
Done, #18049. Code on new master and slow test matrix cut down. Will wait on input from matrix person.